### ISSUE TITLE
General improvements. 

### ISSUE THREAD

**Issue by ITronlegendsoffical** [OffRange/PasswordManager#7]
**please check description.**

**Description:**
**1:  more options to add to password manager.**
- add email option. 
- add website address.
- phone number 

**2: app improvements.**

- face authentication.
- search pre text "search" should be changed to "search passwords"
- import/ export passwords from browsers. 
- login menu feels too bland. Add a logo on top and make the "password manager" text in the top middle.
- add a forget password option. When a user is connected to a special key / email it should give a confirmation email to change your master password.

**Comment by OffRange** [comment]
1. Certainly! Expanding the application to accommodate additional functions for saving data is entirely feasible. I will consider adding it in future versions.

2. 
   + Face authentication is already supported. However, since most Android devices do not have a depth camera, this authentication method is not as secure as Apple's Face ID.
   + Since the application supports more than just passwords, I won't change it to "Search passwords." However, it can be modified to "Search elements" or "Search for elements."
   + The autofill feature on Android partially supports auto-storing passwords (e.g. in other apps where you enter a password). However, most browsers render password fields independently, resulting in the autofill API retrieving passwords as asterisks (*****). See #2
Additionally read the [Android documentation](https://developer.android.com/reference/android/service/autofill/AutofillService#CompatibilityMode).
    > The value of password fields is most likely masked—for example, **** instead of 1234. Hence, you must be careful when using these values to avoid updating the user data with invalid input. For example, when you parse the [FillRequest](https://developer.android.com/reference/android/service/autofill/FillRequest) and detect a password field, you could check if its [input type](https://developer.android.com/reference/android/app/assist/AssistStructure.ViewNode#getInputType()) has password flags and if so, don't add it to the [SaveInfo](https://developer.android.com/reference/android/service/autofill/SaveInfo) object.
   + Adding a logo to the login screen and centering the title is under consideration.
   + I understand your concern about centralized server storage for all accounts. Implementing this feature without compromising security might be challenging. One possibility could be sending a one-time password to the user's email. Nevertheless, this approach poses potential risks, especially since an attacker with access to the unlocked phone could read emails from the user's mail app in most cases. If you have a better and more secure solution, I am open to hearing it. Feel free to create a pull request with your proposed changes, if you wish to contribute.
   + The feature could be implemented in future releases.

**Comment by ITronlegendsoffical** [comment]
Oh about the 3rd importing option I meant you can import CSV files that you can get from chrome/ basically any browser to import the password into the app.
I'll add any other suggestions I get in mind later. 
I'll make a pull request if I get a idea or something lol

**Comment by ITronlegendsoffical** [comment]
The only possible way we could actually store passwords is via a small USB drive that has very hard encryption.

**Comment by OffRange** [comment]
Yes, but why should we use a USB device with strong encryption when all elements (e.g. passwords, credit cards) are already stored encrypted directly on the user's device?

**Comment by ITronlegendsoffical** [comment]
What if the device is stolen? You wouldn't be able to access the password because it isn't Connected to the cloud or anything.  I can keep my USB at home where it won't be stolen. And even it's stolen the theif can't do anything lol because it's encrypted. Im suggesting a USB version of it somehow. I know you could just use import/ export password option but it would be annoying and it would be easier just to make the application somehow find the file automatically.

**Comment by ITronlegendsoffical** [comment]
Also I want to ask a question. Does the password manager support aes 128 or 256?

**Comment by OffRange** [comment]
> What if the device is stolen? You wouldn't be able to access the password because it isn't Connected to the cloud or anything. I can keep my USB at home where it won't be stolen. And even it's stolen the theif can't do anything lol because it's encrypted. Im suggesting a USB version of it somehow. I know you could just use import/ export password option but it would be annoying and it would be easier just to make the application somehow find the file automatically.

How about introducing a new feature that allows users to sync their passwords and other saved elements with a cloud service of their choice? Users would gain the ability to securely store their entire password/element database on their personal cloud storage, such as Google Drive, ensuring easy accessibility on all their devices. Provided users are connected to their Google account on their phone.

> Also I want to ask a question. Does the password manager support aes 128 or 256?

Thank you for bringing up the question!
To better organize and address questions like yours, I have created a dedicated Q&A post in the discussions section of our GitHub repository. You can find it here:: #8
This way, we can keep the discussions organized and ensure that everyone can easily find relevant information.

**Comment by ITronlegendsoffical** [comment]
Yea that's possible. Hacking google accounts is pretty hard nowadays. 
I have more suggestions. 
Password generator when adding passwords. 
Built in 2fa authentication. (This should get a full section probably)
In settings add themes. A AMOLED version, and color picking for devices that don't have material you support or just you want the app not to be the material you wallpaper based color.

**Comment by OffRange** [comment]
A password generator is already integrated.
2FA would require a central server to handle one-time passport details or email sending processes.
I'm already thinking about adding a theme selector for a dark and light mode. However, I won't add a color picker because in my opinion the effort would be greater than the benefit in this app.

**Comment by ITronlegendsoffical** [comment]
Ah I see. Here are some more suggestions. 
# Massive improvements to password manager.
## Section 1 New elements. 
- Router logins
- Insurance details
- ID card information
- Bank details
- Membership details
- Software licence
- passport 
- driving licence
- social security numbers
- Wifi logins 
- App login 
- Utility accounts
- Prescription details
- Investment account details
- Vpn details 
- legal document details. 
- notes
## Section 2 tiles 
### What are tiles?
**Tiles are specific sections in elements.**
Example: You added a password. But you want to add a note to it. So you will click "add a tile" then click notes from a dropdown. Now there will be a note section in the password element. 
Here are some examples of tiles ( I'll probably suggest more in the future)
- password 
- title
- text
- phone number
- credit card number
- note
- image 
- license key 
- IP 
- Port number
- Medical ( basically text tile but named medical ) 
- dates
- date of expiry
- custom tile (basically a text option but changeable tile name) 
**All of these should be removable and addable except title.**
## Section 3 feature improvements. 
- If you added a location/Origin . Add a button next to the
Origin that says (open location in maps) 
Should open in Google maps. 
If possible add a location image to preview the location!
- Add a setting to Wipe all data from the app ( it should ask you 3 times for the master password.) 
- (this connects to date of expiry in section 2) when a element has the date of expiry tile it should appear in highlights when the date of expiry has expired. Also should give a notification that the element has expired.
